assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java

```diff
@@ -883,4 +885,15 @@ private void removeNonAssociatedExternalGradebookEntry(String context, String as
         }
     }
 
+    public void hasRubricSelfReport(Context context, String assignmentId) {
+        try {
+            Optional<ToolItemRubricAssociation> rubricAssociation = rubricsService.getRubricAssociation(RubricsConstants.RBCS_TOOL_ASSIGNMENT, assignmentId);
+            if (rubricAssociation.isPresent() && rubricAssociation.get().getParameter(RubricsConstants.STUDENT_SELF_REPORT)) {
+                context.put("rubricSelfReport", true);
```

Comment:
rubricSelfReport, if you use it more than once, should be in a constant, with a comment about what it means.

Reply:
Thank you, Adrian, for your comments. They are improvements without a doubt.
Right now, "rubricSelfReport", is used only once in a java class.

Reply:
why is this a parameter shouldn't the evaluator be used?

Reply:
In this case, this is not exactly a rubric related parameter but an assignments one. If the assignment is of this type then the velocity will display different things, that's why it's a parameter included in the context.